crates/oci/src/client.rs

```diff
@@ -1108,6 +1218,32 @@ mod test {
                 }]),
                 expected_layer_count: 4,
                 expected_error: None,
+                compose_mode: ComposeMode::Skip,
+            },
+            TestCase {
+                name: "One component layer and one dependency component layer with composition",
+                opts: Some(ClientOpts{content_ref_inline_max_size: 0}),
+                locked_components: from_json!([{
+                "id": "component-with-deps",
+                "source": {
+                    "content_type": "application/wasm",
+                    "source": format!("file://{}", working_dir.path().join("root.wasm").to_str().unwrap()),
+                    "digest": "digest",
+                },
+                "dependencies": {
+                    "test:test/a": {
+                        "source": {
+                            "content_type": "application/wasm",
+                            "source": format!("file://{}", working_dir.path().join("dep_a.wasm").to_str().unwrap()),
+                            "digest": "digest",
+                        },
+                        "export": null,
+                    }
+                }
+                }]),
+                expected_layer_count: 1,
```

Comment:
Does this validate the changes to the lockfile (e.g. component source is now the composed component, component now has no deps)?  I am not sure how practical that is.

Reply:
It does not; i have manually verified that. If you think having that coverage here makes sense I can see about factoring out the "compose" tests to asserts against the final locked manifest layer.

Reply:
Happy to punt on that!